### ISSUE TITLE
Infinite loop when positioning inline image (was: massive CPU load and error with "Java heap space")

### ISSUE THREAD

**Issue by rammpeter** [asciidoctor/asciidoctor-pdf#2193]
Release 2.0.0 eats excessive CPU for longer time and ends up with:
Java heap space
Use --trace to show backtrace

This occurs using jRuby 9.3.4.0 as well as using ruby 3.1.2.

command is
`asciidoctor-pdf -o movex-cdc.pdf movex-cdc.adoc`

The used adoc file can be found here to reproduce the problem: 
https://gitlab.com/otto-group-solution-provider/movex-cdc/-/blob/master/doc/movex-cdc.adoc

With asciidoctor-pdf 1.6.2 it works without problems.

**Comment by mojavelinux** [comment]
The heap space error itself is a red herring. The real problem is that it is getting stuck doing an operation. If I cut the document down to this line, I can reproduce it:

```
image:er_model.svg[format=svg,opts=inline]
```

The hang may have something to do with the logic for resizing an inline image. But that also points us to the actual problem in the document. This should be a block image, not an inline image. If you turn it into an block image, you will be able to convert the document using Asciidoctor PDF 2.

```
image::er_model.svg[format=svg,opts=inline]
```

See https://docs.asciidoctor.org/asciidoc/latest/macros/images/#block-image-macro

I'll have a closer look as to why it is getting stuck when it's an inline image.

**Comment by mojavelinux** [comment]
I figured it out. The text arranger was not able to fit a full height inline image on a new page, and thus would keep trying on the next page after that. I have added a protection against this in general, then tweaked the computation for the positioning of a full height image so that it does fit on the next page.

**Comment by rammpeter** [comment]
@mojavelinux, Thank you very much for your quick response and your useful hints

**Comment by mojavelinux** [comment]
You're welcome! You can now access this fix in the 2.0.1 release, though I still think you'd be better off using a block image in this case.
